vmdb/app/models/mixins/naming_index_mixin.rb

```diff
@@ -0,0 +1,20 @@
+module NamingIndexMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    has_many :naming_indexes, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute", :conditions => {:source => NAMING}
+  end
+
+  def next_naming_index(name)
+    with_lock do
```

Comment:
Different `.lock`.  We want to use the one here: https://github.com/ManageIQ/manageiq/blob/master/vmdb/lib/extensions/ar_lock.rb#L17-L38

On that note, it seems we are overriding the lock method provided by Rails.  I think that was unintentional.  We may want to renaming ours or perhaps get our extension back into Rails in some form, so it isn't confusing, and so it doesn't break anything expecting the Rails interface.  cc @tenderlove @jrafanie